_data-prepper/pipelines/configuration/processors/key-value.md

```diff
@@ -12,18 +12,18 @@ nav_order: 45
-key_value_delimiter_regex| Conditionally | String | A regex specifying the delimiter between a key and a value. Special regex characters such as `[` and `]` must be escaped using `\\`. There is no default value. This option cannot be defined at the same time as `value_split_characters`.
-value_split_characters | Conditionally | String | A string of characters to split between keys and values. Special regex characters such as `[` and `]` must be escaped using `\\`. Default value is `=`. This option cannot be defined at the same time as `key_value_delimiter_regex`.
-non_match_value | No | String | When a key/value cannot be successfully split, the key/value is placed in the `key` field, and the specified value is placed in the value field. Default value is `null`.
-prefix | No | String | A prefix given to all keys. Default value is empty string.
-delete_key_regex | No | String | A regex used to delete characters from the key. Special regex characters such as `[` and `]` must be escaped using `\\`. There is no default value.
-delete_value_regex | No | String | A regex used to delete characters from the value. Special regex characters such as `[` and `]` must be escaped using `\\`. There is no default value.
+Option | Required | Type | Description | Example
+:--- | :--- | :--- | :--- | :---
+source | No | String | The key in the event that is parsed. Default value is `message`. |
+destination | No | String | The destination key for the parsed source output. Outputting the parsed source overwrites the value of the key if it already exists. Default value is `parsed_message` |
+field_delimiter_regex | Conditionally | String | A regex specifying the delimiter between key/value pairs. Special regex characters such as `[` and `]` must be escaped using `\\`. This option cannot be defined at the same time as `field_split_characters`. |
+field_split_characters | Conditionally | String | A string of characters to split between key/value pairs. Special regex characters such as `[` and `]` must be escaped using `\\`. Default value is `&`. This option cannot be defined at the same time as `field_delimiter_regex`. |
+key_value_delimiter_regex| Conditionally | String | A regex specifying the delimiter between a key and a value. Special regex characters such as `[` and `]` must be escaped using `\\`. There is no default value. This option cannot be defined at the same time as `value_split_characters`. |
+value_split_characters | Conditionally | String | A string of characters to split between keys and values. Special regex characters such as `[` and `]` must be escaped using `\\`. Default value is `=`. This option cannot be defined at the same time as `key_value_delimiter_regex`. |
+non_match_value | No | String | When a key/value cannot be successfully split, the key/value is placed in the `key` field, and the specified value is placed in the value field. Default value is `null`. | `key1value1&key2=value2` parses into `{"key1value1": null, "key2": "value2"}`. |
```

Comment:
Include examples section after this table that summarizes examples. Should we include a configuration reference before or after these examples?

Reply:
Example syntax: pipeline configuration, input event, output event.
Refer to this for formatting: https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-plugins/key-value-processor/README.md#basic-usage

Reply:
@asifsmohammed Are there any examples beyond the three that are provided in this doc?

Reply:
@asifsmohammed Thanks for taking another look at this, I'll monitor for updates on the examples as discussed today.